### DISCUSSION TITLE
Trying to create a commit but getting GitlabCreateError: 400 due to incorrect file_path

### DISCUSSION THREAD

**Discussion by gjarzebak95** [python-gitlab/python-gitlab#2358] [Q&A] [answered]
Hey guys I'd really appreciate some help as I've wasted over 2 hours on this supposedly simple issue

I'm trying to create a commit and getting the following error 

raise error(e.error_message, e.response_code, e.response_body) from e
gitlab.exceptions.GitlabCreateError: 400: actions[0][action] is missing, actions[0][action] is invalid, actions[0][file_path] is missing, actions[0][file_path] is invalid

Don't understand is as file_path looks correct to me as well as exists in the repostitory where I'm trying to create the commit

I would really appreciate if somebody could advise me.

Greetings

```
project.commits.create(commit_data)

    commit_data = {
    "branch": branchname,
    "commit_message": "ABCD",
    "actions": [
        actions 
    ]
    }

actions variable printed below

{
"action": "update",
"file_path": "versions.tf",
"content": "terraform {
  required_version = ">= 0.14, < 1.0"
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = ">= 4.19, < 5.0"
    }
  }
}
"
}
```

**Comment by JohnVillalovos** [comment]
If you can provide a minimal reproducer of sample code that would be useful. As at the moment it is more of guess work on my part to figure out the issue.

An example of using `project.commits.create()` is in the functional tests:
https://github.com/python-gitlab/python-gitlab/blob/12aea32d1c0f7e6eac0d19da580bf6efde79d3e2/tests/functional/api/test_repository.py#L95-L105

**Comment by JohnVillalovos** [comment, marked as answer]
So my theory is that the `actions` variable is a string and not a dictionary.

**Reply by JohnVillalovos** [reply]
I tested this and when the `actions` variable was a string I got the same error. So to correct this make sure that the `actions` variable is a dictionary and not a string representation of a dictionary.
